src/homework6/decorator.py

```diff
@@ -0,0 +1,51 @@
+"""Создайте декоратор, который вызывает задекорированную функцию
+пока она не выполнится без исключений (но не более n раз - параметр
+декоратора). Если превышено количество попыток, должно быть возбуждено
+исключение типа TooManyErrors
+
+"""
+
+
+class TooManyErrors(Exception):
+    def __str__(self):
+        return 'TooManyErrors'
+
+
+def decorator(param):
+    def dec(func):
+        def wrapper(*args, **kwargs):
+            errors = param
+            try:
+                while errors:
+                    try:
+                        result = func(*args, **kwargs)
+                        return result
+                    except Exception:
+                        errors -= 1
+                        print('Something wrong')
+                else:
+                    raise TooManyErrors
+            except TooManyErrors:
```

Comment:
I don't understand why should we raise it, than at the same time catch it and then return exception class.